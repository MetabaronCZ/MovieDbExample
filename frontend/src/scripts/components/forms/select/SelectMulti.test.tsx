import { act, render } from '@testing-library/react';
import { describe, expect, it, jest } from '@jest/globals';
import {
  fireEvent,
  getByRole,
  getByTitle,
  getAllByRole,
} from '@testing-library/dom';

import { SelectMulti } from './SelectMulti';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/forms/select/SelectMulti', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(
      <SelectMulti
        value={[1]}
        options={[
          { title: 'Item 1', value: 1 },
          { title: 'Item 2', value: 2 },
        ]}
        onSelect={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();

    const list = getByRole(container, 'list');
    expect(list).toBeInTheDocument();

    const listItems = getAllByRole(list, 'button');
    expect(listItems.length).toEqual(2);

    const selectedItem = listItems[0];
    expect(selectedItem.textContent).toEqual('Item 1');
    expect(selectedItem.nodeName.toLowerCase()).toEqual('button');

    const clearSelectedItem = listItems[1];
    expect(clearSelectedItem.title).toEqual('Odebrat');
    expect(clearSelectedItem.nodeName.toLowerCase()).toEqual('button');

    const clearAllItems = getByTitle(container, 'Odebrat');
    expect(clearAllItems).toBeInTheDocument();
    expect(clearAllItems.nodeName.toLowerCase()).toEqual('button');

    const openButton = getByTitle(container, 'Otevřít');
    expect(openButton).toBeInTheDocument();
    expect(openButton.nodeName.toLowerCase()).toEqual('button');
  });

  it('should not render when no options', async () => {
    const { container } = await waitForComponent(
      <SelectMulti value={[]} options={[]} onSelect={() => null} />,
      { wrapper: TestComponentWrapper },
    );
    expect(container).toMatchSnapshot();
    expect(container.innerHTML).toEqual('');
  });

  it('should throw on invalid value', () => {
    // supress console.error
    jest.spyOn(console, 'error').mockImplementation(() => jest.fn());

    const action = (): void => {
      render(
        <SelectMulti
          value={[100]}
          options={[{ title: 'Item 1', value: 1 }]}
          onSelect={() => null}
        />,
        { wrapper: TestComponentWrapper },
      );
    };
    expect(action).toThrow();

    // restore console.error supression
    jest.restoreAllMocks();
  });

  it('should be openable / closeable', async () => {
    const { container } = await waitForComponent(
      <SelectMulti
        value={[1]}
        options={[
          { title: 'Item 1', value: 1 },
          { title: 'Item 2', value: 2 },
        ]}
        onSelect={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    const handle = getByTitle(container, 'Otevřít');
    expect(handle).toBeInTheDocument();

    let list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);

    // open Select by handle click
    act(() => {
      fireEvent.click(handle);
    });
    expect(container).toMatchSnapshot();

    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(2);

    const selectableItems = getAllByRole(list[1], 'button');
    expect(selectableItems.length).toEqual(1);
    expect(selectableItems[0]).toHaveTextContent('Item 2');

    // close Select by Escape key
    act(() => {
      fireEvent.keyUp(document, { key: 'Escape' });
    });
    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);
  });

  it('should not open when disabled', async () => {
    const { container } = await waitForComponent(
      <SelectMulti
        value={[1]}
        options={[
          { title: 'Item 1', value: 1 },
          { title: 'Item 2', value: 2 },
        ]}
        disabled
        onSelect={() => null}
      />,
      { wrapper: TestComponentWrapper },
    );
    const handle = getByTitle(container, 'Otevřít');
    expect(handle).toBeInTheDocument();

    let list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);

    // try open Select by handle click
    act(() => {
      fireEvent.click(handle);
    });

    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);
  });

  it('should select items', async () => {
    const onSelect = jest.fn();

    const { container } = await waitForComponent(
      <SelectMulti
        value={[1]}
        options={[
          { title: 'Item 1', value: 1 },
          { title: 'Item 2', value: 2 },
        ]}
        onSelect={onSelect}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(onSelect).not.toHaveBeenCalled();

    const handle = getByTitle(container, 'Otevřít');

    let list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);

    const selected = getAllByRole(list[0], 'listitem');
    expect(selected.length).toEqual(1);
    expect(selected[0].textContent).toEqual('Item 1');

    // open Select by handle click
    act(() => {
      fireEvent.click(handle);
    });
    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(2);

    const selectableItems = getAllByRole(list[1], 'button');
    expect(selectableItems.length).toEqual(1);
    expect(selectableItems[0]).toHaveTextContent('Item 2');

    // click Item 2
    act(() => {
      fireEvent.click(selectableItems[0]);
    });
    expect(onSelect).toHaveBeenCalledTimes(1);
    expect(onSelect).toHaveBeenCalledWith([1, 2]); // Item 1 + Item 2 values

    // Select is still opened for next selection
    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(2);
  });

  it('should not select disabled item', async () => {
    const onSelect = jest.fn();

    const { container } = await waitForComponent(
      <SelectMulti
        value={[1]}
        options={[
          { title: 'Item 1', value: 1 },
          { title: 'Item 2', value: 2, disabled: true },
        ]}
        onSelect={onSelect}
      />,
      { wrapper: TestComponentWrapper },
    );
    expect(onSelect).not.toHaveBeenCalled();

    const handle = getByTitle(container, 'Otevřít');

    let list = getAllByRole(container, 'list');
    expect(list.length).toEqual(1);

    const selected = getAllByRole(list[0], 'listitem');
    expect(selected.length).toEqual(1);
    expect(selected[0].textContent).toEqual('Item 1');

    // open Select by handle click
    act(() => {
      fireEvent.click(handle);
    });
    expect(container).toMatchSnapshot();

    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(2);

    const selectableItems = getAllByRole(list[1], 'button');
    expect(selectableItems.length).toEqual(1);
    expect(selectableItems[0]).toHaveTextContent('Item 2');

    // click Item 2
    act(() => {
      fireEvent.click(selectableItems[0]);
    });
    expect(onSelect).not.toHaveBeenCalled();

    // Select is still opened
    list = getAllByRole(container, 'list');
    expect(list.length).toEqual(2);
  });
});
