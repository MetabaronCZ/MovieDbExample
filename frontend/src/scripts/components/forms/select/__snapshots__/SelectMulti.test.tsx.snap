// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`components/forms/select/SelectMulti should be openable / closeable 1`] = `
.c7 {
  display: flex;
  flex-direction: row;
  gap: 0px;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c9 {
  position: absolute;
  top: calc(100% + 4px);
  left: 0;
  width: 240px;
  overflow: hidden;
  z-index: 1000;
  background: #fff;
  border: 1px solid #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

.c10 {
  list-style-type: none;
  overflow-y: auto;
  max-height: 360px;
}

.c11 {
  border-bottom: 1px solid #dae0e3;
}

.c11:last-child {
  border-bottom: none;
}

.c12 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  width: 100%;
  padding: 4px 8px;
  background: #fff;
  white-space: nowrap;
  text-align: left;
}

.c12:hover {
  background-color: #f2f7fa;
}

.c12:focus {
  outline: 1px dashed #002d56;
  outline-offset: -1px;
}

.c12:disabled {
  color: #98c3d9;
}

.c8 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c8:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c8:focus {
  outline: 1px dashed #002d56;
}

.c2 {
  list-style-type: none;
  flex: 1;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  gap: 4px;
  padding: 4px 0;
}

.c3 {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1px;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0 8px;
  background: #98c3d9;
  white-space: nowrap;
}

.c5:hover:not(:disabled) {
  background-color: #0af;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  cursor: initial;
}

.c6 {
  width: 24px;
  height: 24px;
  padding: 0;
}

.c0 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
  height: auto;
  gap: 8px;
  min-height: 40px;
  padding-right: 4px;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
        >
          <button
            class="c4 c5"
            type="button"
          >
            Item 1
          </button>
          <button
            class="c4 c5 c6"
            title="Odebrat"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-sm "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="c7"
      >
        <button
          class="c4 c8"
          title="Zrušit"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark fa-lg "
            color="#002d56"
            data-icon="xmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="c4 c8"
          title="Zavřít"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-up fa-lg "
            color="#002d56"
            data-icon="angle-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c9"
    >
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          <button
            class="c4 c12"
            type="button"
          >
            Item 2
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`components/forms/select/SelectMulti should not render when no options 1`] = `<div />`;

exports[`components/forms/select/SelectMulti should not select disabled item 1`] = `
.c7 {
  display: flex;
  flex-direction: row;
  gap: 0px;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c9 {
  position: absolute;
  top: calc(100% + 4px);
  left: 0;
  width: 240px;
  overflow: hidden;
  z-index: 1000;
  background: #fff;
  border: 1px solid #002d56;
  box-shadow: 1px 1px 4px #dae0e3;
}

.c10 {
  list-style-type: none;
  overflow-y: auto;
  max-height: 360px;
}

.c11 {
  border-bottom: 1px solid #dae0e3;
}

.c11:last-child {
  border-bottom: none;
}

.c12 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  width: 100%;
  padding: 4px 8px;
  background: #fff;
  white-space: nowrap;
  text-align: left;
}

.c12:hover {
  background-color: #f2f7fa;
}

.c12:focus {
  outline: 1px dashed #002d56;
  outline-offset: -1px;
}

.c12:disabled {
  color: #98c3d9;
}

.c8 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c8:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c8:focus {
  outline: 1px dashed #002d56;
}

.c2 {
  list-style-type: none;
  flex: 1;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  gap: 4px;
  padding: 4px 0;
}

.c3 {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1px;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0 8px;
  background: #98c3d9;
  white-space: nowrap;
}

.c5:hover:not(:disabled) {
  background-color: #0af;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  cursor: initial;
}

.c6 {
  width: 24px;
  height: 24px;
  padding: 0;
}

.c0 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
  height: auto;
  gap: 8px;
  min-height: 40px;
  padding-right: 4px;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
        >
          <button
            class="c4 c5"
            type="button"
          >
            Item 1
          </button>
          <button
            class="c4 c5 c6"
            title="Odebrat"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-sm "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="c7"
      >
        <button
          class="c4 c8"
          title="Zrušit"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark fa-lg "
            color="#002d56"
            data-icon="xmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="c4 c8"
          title="Zavřít"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-up fa-lg "
            color="#002d56"
            data-icon="angle-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 137.4c12.5-12.5 32.8-12.5 45.3 0l160 160c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L224 205.3 86.6 342.6c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l160-160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="c9"
    >
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          <button
            class="c4 c12"
            disabled=""
            type="button"
          >
            Item 2
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`components/forms/select/SelectMulti should render properly 1`] = `
.c7 {
  display: flex;
  flex-direction: row;
  gap: 0px;
}

.c4 {
  display: block;
  border: none;
  background: transparent;
  outline: none;
  cursor: pointer;
}

.c4:disabled {
  cursor: not-allowed;
}

.c8 {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 32px;
  height: 32px;
}

.c8:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c8:focus {
  outline: 1px dashed #002d56;
}

.c2 {
  list-style-type: none;
  flex: 1;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  gap: 4px;
  padding: 4px 0;
}

.c3 {
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 1px;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0 8px;
  background: #98c3d9;
  white-space: nowrap;
}

.c5:hover:not(:disabled) {
  background-color: #0af;
}

.c5:focus {
  outline: 1px dashed #002d56;
}

.c5:disabled {
  cursor: initial;
}

.c6 {
  width: 24px;
  height: 24px;
  padding: 0;
}

.c0 {
  position: relative;
  display: inline-block;
  width: 100%;
}

.c1 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  gap: 12px;
  line-height: 1;
  font-weight: 700;
  height: 40px;
  padding: 0 16px;
  padding-left: 12px;
  padding-right: 12px;
  border: 2px solid #002d56;
  background: #fff;
  width: 100%;
  height: 40px;
  gap: 8px;
  padding-left: 12px;
  padding-right: 12px;
  height: auto;
  gap: 8px;
  min-height: 40px;
  padding-right: 4px;
}

.c1:hover:not(:disabled) {
  background-color: #f2f7fa;
}

.c1:focus {
  outline: 1px dashed #002d56;
}

.c1:disabled {
  color: #98c3d9;
  border-color: #98c3d9;
  background-color: #fff;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <ul
        class="c2"
      >
        <li
          class="c3"
        >
          <button
            class="c4 c5"
            type="button"
          >
            Item 1
          </button>
          <button
            class="c4 c5 c6"
            title="Odebrat"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark fa-sm "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
      </ul>
      <div
        class="c7"
      >
        <button
          class="c4 c8"
          title="Zrušit"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-xmark fa-lg "
            color="#002d56"
            data-icon="xmark"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
              fill="currentColor"
            />
          </svg>
        </button>
        <button
          class="c4 c8"
          title="Otevřít"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angle-down fa-lg "
            color="#002d56"
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M201.4 374.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 306.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
